torchvision/models/detection/roi_heads.py

```diff
@@ -525,6 +527,15 @@ def forward(self, features, proposals, image_shapes, targets=None):
             image_shapes (List[Tuple[H, W]])
             targets (List[Dict])
         """
+        if targets is not None:
+            for t in targets:
+                assert t["boxes"].dtype == torch.float32, 'target boxes must of float type'
+                assert t["labels"].dtype == torch.int64, 'target labels must of int64 type'
+                if self.has_mask:
+                    assert t["masks"].dtype == torch.uint8, 'target masks must of uint8 type'
```

Comment:
Do the masks *need* to be `uint8`? I thought that the code worked as well for other types, as the only place where it is used in the model we perform a cast to `float`?

Reply:
I was following the [tutorial](https://colab.research.google.com/github/pytorch/vision/blob/temp-tutorial/tutorials/torchvision_finetuning_instance_segmentation.ipynb), where the dtype of mask was specified `uint8`. Should I remove that check?

Reply:
I don't think having the masks being `uint8` is a hard-restriction, so yes, maybe remove this check for now

Reply:
Done.